### DISCUSSION TITLE
How to "excel" mesa properly

### DISCUSSION THREAD

**Discussion by KgPergher98** [mesa/mesa#1047] [Q&A] [answered]
Hi guys, I'm new on Mesa and quite new on Python too (please be patient with me :) ) and I'm starting to develop some agent-based models. I'm struggling a bit with the API's reference and Docs and I'm not being able to find more detailed references to some classes methods like MultiGrid and SimpleGrid. 
If someone could provide me a path to solve it by my own I'll be so glad. Thanks!

**Comment by rht** [comment, marked as answer]
There is a problem with the documentation generation. https://mesa.readthedocs.io/en/master/apis/space.html is empty, while https://github.com/projectmesa/mesa/blob/af4de1d31d62b3e8dd71606bd94ed643902821e3/mesa/space.py still has some docstrings.
I think the Mypy annotation broke the doc. I will try to fix it.

In the meantime, you can just read the docstring in https://github.com/projectmesa/mesa/blob/af4de1d31d62b3e8dd71606bd94ed643902821e3/mesa/space.py. It's not as nicely formatted, but it has the info you need.

**Reply by rht** [reply]
This version, https://mesa.readthedocs.io/en/doc_builds/apis/space.html, seems to render properly. But it is not the latest version (https://github.com/projectmesa/mesa/tree/doc_builds), is 97 commits behind from main.

**Reply by jackiekazil** [reply]
@tpike3 was this the issue you brought up tonight? And were you looking into it?

**Reply by KgPergher98** [reply]
Thanks for you support, it should help a lot as I see! 👍🏻

**Reply by rht** [reply]
I have looked at the commits in docs/ that are new since the doc_builds branch. This commit is the only one that is likely to cause the problem: https://github.com/projectmesa/mesa/commit/c9434fd14b8edbc285f29a12c582780b26b9abea#diff-46b42b4229cd7a39c564e780bb665a8bde4fdf722007e8473f167fe53ed4b995. The automodule should probably just be `space` instead of `mesa.space`. @tpike3 @jackiekazil want to test this hypothesis in a test branch, and see it in RTD?

For context, `datacollection` doesn't have the `mesa.` prefix: https://github.com/projectmesa/mesa/blob/main/docs/apis/datacollection.rst, but `time` does: https://github.com/projectmesa/mesa/blob/main/docs/apis/time.rst.

**Reply by tpike3** [reply]
@rht Will do. I will test ```space``` instead of ```mesa.space``` to see if it resolves the issue. Thanks for tracking that done!
